tests/py/test_email.py

```diff
@@ -219,6 +220,13 @@ def test_html_escaping(self):
         assert 'foo&#39;bar' in last_email['html']
         assert '&#39;' not in last_email['text']
 
+    @mock.patch('gratipay.models.participant.Participant.send_email')
+    def test_emails_with_plus_are_linked_properly(self, send_email):
+        send_email.return_value = 1
+        self.alice.add_email("foo+bar@example.com")
+        link = send_email.args[1]
+        assert link.startswith('/~alice/emails/verify.html?email=foo%2Bbar%40example.com&nonce=')
+
```

Comment:
>  Something I don't know about `mock` yet?

Yup, you called it: `link` is a mock, which thinks everything is a-okay. Guido [pointed](https://twitter.com/gvanrossum/status/648617596011417600) this up recently.

```
(Pdb) l
246         def test_emails_with_plus_are_linked_properly(self, send_email):
247             send_email.return_value = 1
248             self.alice.add_email("foo+bar@example.com")
249             link = send_email.args[1]
250             import pdb; pdb.set_trace()
251  ->         assert link.startswith('/~alice/emails/verify.html?email=foo%2Bbar%40example.com&nonce=')
252     
253         def test_can_dequeue_an_email(self):
254             larry = self.make_participant('larry', email_address='larry@example.com')
255             larry.queue_email("verification")
256     
(Pdb) pp link
<MagicMock name=u'send_email.args.__getitem__()' id='4383627408'>
(Pdb)
```